### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by IjayAbby**
Learning Objectives.

- Use JavaScript to make websites dynamic and build basic single page apps.
- Use ES6 syntax.
- Use ES6 modules.
- Use callbacks and promises.
- Use webpack.
- Apply JavaScript best practices and language style guides in code.
- Use AAA pattern for unit tests.
- Write units tests for a JavaScript app.
- Follow Gitflow.
- Solve simple git conflicts.
- Send and receive data from an API.
- Use API documentation.
- Understand and use JSON.
- Make JavaScript code synchronous.
- Perform a code review for a team member.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello Team 👋,

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._

Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1,2 +1,117 @@
...
+- Follow Gitflow.
+- Solved simple git conflicts.
+- Sent and receive data from an API.
+- Used API documentation.
+- Understood and Used JSON.
+- Made JavaScript code synchronous.
+- Performed a code review for a team member.
+
+## Live Demo
+Click [here](https://francisuloko.github.io/javascript-capstone/dist/) to visit the app
```
Comment (mikenath223) [inline README.md:31]: - [ ] The comment functionality doesn't work on the live version. Kindly check it.

File: src/comment.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import API from './api.js';
+
+const api = new API();
+
+export function countComment(res) {
+  if (res.length > 0) {
+    return res.length;
+  }
+  return 0;
+}
```
Comment (mikenath223) [inline src/comment.js]: - [ ] Kindly change this and all other regular functions into es6 arrow functions. As required in the [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md#:~:text=If%20you%20use%20ES6%2C%20use%20object%20destructuring%20to%20get%20the%20values%20from%20an%20object.%20This%20way%20you%20can%20avoid%20repeating%20a%20lot%20of%20code.) it's important to use es6 syntax.

File: src/utilities.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import Comment from './comment.js';
+
+const api = new API();
+const comment = new Comment();
+
+export default function popUp(e) {
+  const mealID = e.target.dataset.id;
+  const modal = document.getElementById('modal');
+  modal.classList.toggle('hide');
+  modal.innerHTML = '';
```
Comment (mimipeshy) [inline src/utilities.js]: - [ ] Like the previous TSE mentioned, kindly change this and all other `regular functions` into es6 `arrow functions`.

**Review by mimipeshy — CHANGES_REQUESTED** [review]
## Hi @Team 👋🏾 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](<https://microverse.zendesk.com/hc/en-us/articles/1500004088561>)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](<https://airtable.com/shrQAqnBwek5a0O0s>)._

**Review by mimipeshy — APPROVED** [review]
## Hi @Team 👋🏾 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](<https://microverse.zendesk.com/hc/en-us/articles/1500004088561>)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](<https://airtable.com/shrQAqnBwek5a0O0s>)._

**Comment by francisuloko** [comment]
Hi @mimipeshy @mikenath223 

Thank you for reviewing and approving our code.

Happy Coding :computer:

**Pull request merged**